### PR TITLE
SLING-7625 refactor of the Default Get Servlet bundle

### PR DESCRIPTION
**Pull request by JEBailey**
In general
Remove all uses of deprecated API's
Remove all methods that aren't being used
Re-organize GET servlets to be in one location
Rename objects to better represent their purpose
Reduce memory consumption by removing unnecessary Object creation

I can go into specific details on any specific change

### CODE REVIEW

File: src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java [hunk 1]
```diff
@@ -188,31 +189,18 @@ protected void deactivate() {
         this.indexFiles = null;
     }
 
-    private Servlet getDefaultRendererServlet(final String type) {
-        Servlet servlet = null;
+    private Renderer getDefaultRendererServlet(final String type) {
+    	Renderer servlet = null;
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java]: Renaming the variable here would make sense as well.

File: src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java [hunk 2]
```diff
@@ -223,7 +211,7 @@ public void init() throws ServletException {
 
         // use the servlet for rendering StreamRendererServlet.EXT_RES as the
         // streamer servlet
-        Servlet streamerServlet = getDefaultRendererServlet(EXT_RES);
+        Renderer streamerServlet = getDefaultRendererServlet(EXT_RES);
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java]: Renaming the variable here would make sense as well.

File: src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java [hunk 3]
```diff
@@ -291,7 +279,7 @@ protected void doGet(SlingHttpServletRequest request,
                 request.getResource().getPath(), "No resource found");
         }
 
-        Servlet rendererServlet;
+        Renderer rendererServlet;
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java]: Renaming the variable here would make sense as well.

Comment (JEBailey) [reply src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java]: good point

**Pull request merged**